applications/RANSApplication/python_scripts/formulations/turbulence_models/scalar_turbulence_model_rans_formulation.py

```
@@ -0,0 +1,208 @@
+            return True
+
+        return False
+
+    def GetStrategy(self):
+        return self.solver
+
+    def SetTimeSchemeSettings(self, settings):
+        if (settings.Has("scheme_type")):
+            scheme_type = settings["scheme_type"].GetString()
+            if (scheme_type == "steady"):
+                self.is_steady_simulation = True
+            elif (scheme_type == "bdf2" or scheme_type == "bossak"):
+                self.is_steady_simulation = False
+            else:
+                raise Exception(
+                    "Only \"steady\", \"bdf2\" and \"bossak\" scheme types supported. [ scheme_type = \""
+                    + scheme_type + "\" ]")
+        else:
+            raise Exception(
+                "\"scheme_type\" is missing in time scheme settings")
+
+    def GetMaxCouplingIterations(self):
+        return "N/A"
+
+    def GetModelPart(self):
+        return self.turbulence_model_part
+
+    def SetStabilizationMethod(self, stabilization_method):
+        self.element_name = self.GetElementNamePrefix()
+        if (stabilization_method == "algebraic_flux_corrected"):
+            self.element_name = self.element_name + "AFC"
+            self.scheme_type = self._CreateAlgebraicFluxCorrectedSteadyScalarScheme
+        elif (stabilization_method == "residual_based_flux_corrected"):
+            self.element_name = self.element_name + "RFC"
+            self.scheme_type = GetKratosObjectType("SteadyScalarScheme")
+        elif (stabilization_method == "non_linear_cross_wind_dissipation"):
+            self.element_name = self.element_name + "CWD"
+            self.scheme_type = GetKratosObjectType("SteadyScalarScheme")
+        else:
+            raise Exception("Unsupported stabilization method")
+
+    def SetWallFunctionSettings(self, settings):
+        self.condition_name = self.GetConditionNamePrefix()
+
+        if (self.condition_name != ""):
```

I'd throw an error if this condition is not satisfied.

---

I am afraid it is not possible, because there are some formulations which does not use any conditions. So, the `GetConditionNamePrefix` will return blank string in this case. (In this case, duplicated model part will not have any conditions)